android/libraries/rib-base/src/main/java/com/badoo/ribs/core/routing/configuration/ConfigurationCommandCreator.kt

```diff
@@ -0,0 +1,205 @@
+package com.badoo.ribs.core.routing.configuration
+
+import android.os.Parcelable
+import com.badoo.ribs.core.routing.configuration.ConfigurationCommand.SingleConfigurationCommand.Activate
+import com.badoo.ribs.core.routing.configuration.ConfigurationCommand.SingleConfigurationCommand.Add
+import com.badoo.ribs.core.routing.configuration.ConfigurationCommand.SingleConfigurationCommand.Deactivate
+import com.badoo.ribs.core.routing.configuration.ConfigurationCommand.SingleConfigurationCommand.Remove
+import com.badoo.ribs.core.routing.configuration.ConfigurationKey.Content
+import com.badoo.ribs.core.routing.configuration.ConfigurationKey.Overlay
+import com.badoo.ribs.core.routing.configuration.ConfigurationKey.Overlay.Key
+import com.badoo.ribs.core.routing.configuration.feature.BackStackElement
+import com.badoo.ribs.core.routing.configuration.feature.BackStackFeature
+import io.reactivex.Observable
+import java.lang.Math.min
+
+/**
+ * Takes the state emissions from [BackStackFeature], and translates them to a stream of
+ * [ConfigurationCommand]s.
+ *
+ * @see [ConfigurationCommandCreator.diff]
+ */
+internal fun <C : Parcelable> BackStackFeature<C>.commands(): Observable<ConfigurationCommand<C>> =
```

Comment:
maybe it should be defined as an extension property?
`val BackStackFeature<C>.commands: ...`

Reply:
Done: keeping it as fun + renaming to `.toCommands()` as discussed